google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -87,6 +88,12 @@ def insecure?
           credentials == :this_channel_is_insecure
         end
 
+        def async_publisher
+          return mocked_async_publisher if mocked_async_publisher
```

Comment:
This is a hook used by our tests. This object is not part of the public API exposed to the users.